// Copyright 2024 Google LLC. All Rights Reserved.
// 
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
// 
//     http://www.apache.org/licenses/LICENSE-2.0
// 
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
syntax = "proto3";
// All generated protos should be opaque, with "xxx_hidden_" prepended to their field names.


package dcl;

import "proto/connector/sdk.proto";
import "proto/empty.proto";


enum CloudresourcemanagerBetaProjectLifecycleStateEnum {
  CloudresourcemanagerBetaProjectLifecycleStateEnumNO_VALUE_DO_NOT_USE = 0;
  CloudresourcemanagerBetaProjectLifecycleStateEnumLIFECYCLE_STATE_UNSPECIFIED = 1;
  CloudresourcemanagerBetaProjectLifecycleStateEnumACTIVE = 2;
  CloudresourcemanagerBetaProjectLifecycleStateEnumDELETE_REQUESTED = 3;
  CloudresourcemanagerBetaProjectLifecycleStateEnumDELETE_IN_PROGRESS = 4;
}

message CloudresourcemanagerBetaProject {
  map<string, string> labels = 1;
  CloudresourcemanagerBetaProjectLifecycleStateEnum lifecycle_state = 2;
  string displayName = 3;
  string parent = 4;
  string name = 5;
  int64 project_number = 6;
}

message ApplyCloudresourcemanagerBetaProjectRequest {
  CloudresourcemanagerBetaProject resource = 1;
  repeated LifecycleDirective lifecycle_directives = 2;
  string service_account_file = 3;
}

message DeleteCloudresourcemanagerBetaProjectRequest {
  string service_account_file = 1;
  CloudresourcemanagerBetaProject resource = 2;
}

message ListCloudresourcemanagerBetaProjectRequest {
  string service_account_file = 1;
  string Parent = 2;
}

message ListCloudresourcemanagerBetaProjectResponse {
  repeated CloudresourcemanagerBetaProject items = 1;
}

service CloudresourcemanagerBetaProjectService {
  rpc ApplyCloudresourcemanagerBetaProject(ApplyCloudresourcemanagerBetaProjectRequest) returns (CloudresourcemanagerBetaProject);
  rpc DeleteCloudresourcemanagerBetaProject(DeleteCloudresourcemanagerBetaProjectRequest) returns (google.protobuf.Empty);
  rpc ListCloudresourcemanagerBetaProject(ListCloudresourcemanagerBetaProjectRequest) returns (ListCloudresourcemanagerBetaProjectResponse);
}
